src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -631,7 +631,11 @@ void GcsActorManager::OnNodeDead(const ClientID &node_id) {
 
 void GcsActorManager::ReconstructActor(const ActorID &actor_id, bool need_reschedule) {
   auto &actor = registered_actors_[actor_id];
-  RAY_CHECK(actor != nullptr);
+  // If the owner and this actor is dead at the same time, the actor
+  // could've been destroyed and dereigstered before reconstruction.
+  if (actor == nullptr) {
+    return;
```

Comment:
RAY_LOG(WARNING) << "Actor is destroyed before reconstruction, actor id = "
                     << actor_id;

Reply:
Not sure if this should be warning. This is not an unexpected scenario nor something that makes can cause inconsistent states. Maybe INFO is enough here? What's the reasoning behind WARNING?

Reply:
INFO is ok.